File: server/src/main/java/ua/com/alevel/service/educationalprocess/CourseService.java
package ua.com.alevel.service.educationalprocess;

import org.springframework.stereotype.Service;
import ua.com.alevel.persistence.crudhelper.CrudRepositoryHelper;
import ua.com.alevel.persistence.entity.educationalprocess.Course;
import ua.com.alevel.persistence.repository.educationalprocess.CourseRepository;
import ua.com.alevel.service.AbstractService;

@Service
public class CourseService extends AbstractService<Course, CourseRepository> {

    private final CourseRepository courseRepository;
    private final CrudRepositoryHelper<Course, CourseRepository> crudRepositoryHelper;

    public CourseService(CourseRepository courseRepository, CrudRepositoryHelper<Course, CourseRepository> crudRepositoryHelper) {
        super(courseRepository, crudRepositoryHelper);
        this.courseRepository = courseRepository;
        this.crudRepositoryHelper = crudRepositoryHelper;
    }
}

File: client-admin/src/app/shared/services/educationalprocess/course.service.ts
import {Injectable} from '@angular/core';
import {AbstractService} from "../abstract.service";
import {HttpClient} from "@angular/common/http";
import {Course} from '../../models/educationalprocess/course.model';

@Injectable({
  providedIn: 'root'
})
export class CourseService extends AbstractService<Course> {

  override resourceUrl = '/api/courses';

  constructor(public override httpClient: HttpClient) {
    super(httpClient);
  }
}
